packages/typespec-ts/src/modelUtils.ts

```diff
@@ -89,21 +94,38 @@ export function getSchemaForType(
     return builtinType;
   }
   if (type.kind === "Model") {
-    const schema = getSchemaForModel(program, type, usage, needRef) as any;
+    if (type.name === "" && type.templateMapper && type.templateMapper.args.length > 0) {
+      for(const temp of type.templateMapper.args) {
+        if (temp.kind === "Model" ) {
+          if (temp.name === "") {
+            continue;
+          }
+          type = temp;
+          break;
+        }
+      }
+    }
```

Comment:
for some reason getEffectiveModel doesn't return the empty object model with model name, I added the logic here to get the correct templateArguments.